Source/Core/Duality/Drawing/DrawDevice.cs

```diff
@@ -480,13 +480,17 @@ public BatchInfo RentMaterial()
 			if (vertexCount > vertexBuffer.Length) vertexCount = vertexBuffer.Length;
 			if (material == null) material = Material.Checkerboard.Res.Info;
 
+			// Move the added vertices to an internal shared buffer
+			VertexSlice<T> slice = this.drawVertices.Rent<T>(vertexCount);
+			Array.Copy(vertexBuffer, 0, slice.Data, slice.Offset, slice.Length);
+
```

Comment:
👍  for just moving the copy a bit earlier, probably the best approach to address this.